File: engine/engine-model/src/main/scala/tools/packet/serverlist/ServerList.scala
package tools.packet.serverlist

import net.opcodes.SendOpcode
import tools.packet.{ChannelLoad, PacketInput}

class ServerList(private var _serverId: Int, private var _serverName: String, private var _flag: Int,
                 private var _eventMsg: String, private var _channelLoad: java.util.List[ChannelLoad]) extends PacketInput {
  def serverId: Int = _serverId

  def serverName: String = _serverName

  def flag: Int = _flag

  def eventMsg: String = _eventMsg

  def channelLoad: java.util.List[ChannelLoad] = _channelLoad

  override def opcode(): SendOpcode = SendOpcode.SERVER_LIST
}

File: engine/engine-model/src/main/java/tools/packet/serverlist/ServerList.java
package tools.packet.serverlist;

import java.util.List;

import net.opcodes.SendOpcode;
import tools.packet.ChannelLoad;
import tools.packet.PacketInput;

public record ServerList(Integer serverId, String serverName, Integer flag, String eventMsg,
                         List<ChannelLoad> channelLoad) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SERVER_LIST;
   }
}